dotnet/src/Connectors/Connectors.Memory.Sqlite/SqliteVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,795 @@
+
+    /// <summary>Optional configuration options for this class.</summary>
+    private readonly SqliteVectorStoreRecordCollectionOptions<TRecord> _options;
+
+    /// <summary>The mapper to use when mapping between the consumer data model and the SQLite record.</summary>
+    private readonly IVectorStoreRecordMapper<TRecord, Dictionary<string, object?>> _mapper;
+
+    /// <summary>The default options for vector search.</summary>
+    private static readonly VectorSearchOptions s_defaultVectorSearchOptions = new();
+
+    /// <summary>Command builder for queries in SQLite database.</summary>
+    private readonly SqliteVectorStoreCollectionCommandBuilder _commandBuilder;
+
+    /// <summary>Contains helpers for reading vector store model properties and their attributes.</summary>
+    private readonly VectorStoreRecordPropertyReader _propertyReader;
+
+    /// <summary>Flag which indicates whether vector properties exist in the consumer data model.</summary>
+    private readonly bool _vectorPropertiesExist;
+
+    /// <summary>Collection of properties to operate in SQLite data table.</summary>
+    private readonly Lazy<List<VectorStoreRecordProperty>> _dataTableProperties;
+
+    /// <summary>Collection of properties to operate in SQLite vector table.</summary>
+    private readonly Lazy<List<VectorStoreRecordProperty>> _vectorTableProperties;
+
+    /// <summary>Collection of property names to operate in SQLite data table.</summary>
+    private readonly Lazy<List<string>> _dataTableStoragePropertyNames;
+
+    /// <summary>Collection of property names to operate in SQLite vector table.</summary>
+    private readonly Lazy<List<string>> _vectorTableStoragePropertyNames;
+
+    /// <summary>Table name in SQLite for data properties.</summary>
+    private readonly string _dataTableName;
+
+    /// <summary>Table name in SQLite for vector properties.</summary>
+    private readonly string _vectorTableName;
+
+    /// <inheritdoc />
+    public string CollectionName { get; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SqliteVectorStoreRecordCollection{TRecord}"/> class.
+    /// </summary>
+    /// <param name="connection"><see cref="DbConnection"/> that will be used to manage the data in SQLite.</param>
+    /// <param name="collectionName">The name of the collection/table that this <see cref="SqliteVectorStoreRecordCollection{TRecord}"/> will access.</param>
+    /// <param name="options">Optional configuration options for this class.</param>
+    public SqliteVectorStoreRecordCollection(
+        DbConnection connection,
+        string collectionName,
+        SqliteVectorStoreRecordCollectionOptions<TRecord>? options = default)
+    {
+        // Verify.
+        Verify.NotNull(connection);
+        Verify.NotNullOrWhiteSpace(collectionName);
+        VectorStoreRecordPropertyVerification.VerifyGenericDataModelKeyType(typeof(TRecord), customMapperSupplied: false, SqliteConstants.SupportedKeyTypes);
```

Comment:
`customMapperSupplied` should be checking if a custom mapper was supplied, instead of being hardcoded, otherwise it'll throw for cases where one was supplied and the generic data model is used.

Reply:
Good catch, thanks!